MathLib/LinAlg/PETSc/PETScLinearSolver.cpp

```diff
@@ -63,7 +63,8 @@ PETScLinearSolver::PETScLinearSolver(const std::string /*prefix*/,
         KSPSetOptionsPrefix(_solver, prefix.c_str());
     }
 
-    KSPSetFromOptions(_solver); // set running time option
+    KSPSetFromOptions(_solver); // set run-time options
+    KSPSetInitialGuessNonzero(_solver, PETSC_TRUE);
```

Comment:
Keep only this one, not before solve. Move before setting of the other options. (endJ)